intake_esm/cesm.py

```diff
@@ -342,51 +327,70 @@ def results(self):
-        url = self.urlpath
-        kwargs = self._kwargs
+        kwargs = self.kwargs
 
-        if len(self.query_results) == 0:
-            raise ValueError('query results are empty')
+        if self.query_results.empty:
+            raise ValueError(f'Query={self.query} returned empty results')
 
         query = dict(self.query)
-        if '*' in url or isinstance(url, list):
-            if 'concat_dim' not in kwargs.keys():
-                kwargs.update(concat_dim=self.concat_dim)
-            if self.pattern:
-                kwargs.update(preprocess=self._add_path_to_ds)
-
-            ensembles = self.query_results.ensemble.unique()
-            variables = self.query_results.variable.unique()
-
-            ds_ens_list = []
-            for ens_i in ensembles:
-                query['ensemble'] = ens_i
-
-                ds_var_list = []
-                for var_i in variables:
-
-                    query['variable'] = var_i
-                    urlpath_ei_vi = get_subset(
-                        self.collection_name,
-                        self.collection_type,
-                        query,
-                        order_by=['sequence_order', 'files'],
-                    ).files.tolist()
-
-                    dsets = [
-                        aggregate.open_dataset(url, data_vars=[var_i]) for url in urlpath_ei_vi
-                    ]
-                    ds_var_i = aggregate.concat_time_levels(dsets)
-                    ds_var_list.append(ds_var_i)
-
-                ds_ens_i = aggregate.merge(ds_var_list)
-                ds_ens_list.append(ds_ens_i)
-
-            self._ds = aggregate.concat_ensembles(ds_ens_list, member_ids=ensembles, join='outer')
-        else:
-            self._ds = xr.open_dataset(url, chunks=self.chunks, **kwargs)
+
+        if 'decode_times' not in kwargs.keys():
+            kwargs.update(decode_times=False)
+        if 'time_coord_name' not in kwargs.keys():
+            kwargs.update(time_coord_name='time')
+        if 'ensemble_dim_name' not in kwargs.keys():
+            kwargs.update(ensemble_dim_name='member_id')
+        if 'chunks' not in kwargs.keys():
+            kwargs.update(chunks=None)
+
+        ensembles = self.query_results.ensemble.unique()
+        variables = self.query_results.variable.unique()
+
+        ds_ens_list = []
+        for ens_i in ensembles:
+            query['ensemble'] = ens_i
+
+            ds_var_list = []
+            for var_i in variables:
+
+                query['variable'] = var_i
+                urlpath_ei_vi = get_subset(
+                    self.collection_name,
+                    self.collection_type,
+                    query,
+                    order_by=['sequence_order', 'file_fullpath'],
+                )['file_fullpath'].tolist()
+
+                dsets = [
+                    aggregate.open_dataset(
+                        url, data_vars=[var_i], decode_times=kwargs['decode_times']
+                    )
```

Comment:
What about the chunk argument?

Reply:
This was a tricky case to handle due to the fact that a user may provide `chunks` argument with dimensions that are not available at this stage. For instance, if the user provides `chunks={'member_id' : 2}`, and we try to pass this to `aggregate.open_dataset()`, this function will raise an error due to absence of `member_id` in the dataset. Because of this, I decided to wait until after the final concatenation to chunk the dataset into desired chunks when we are 100% sure that all dimensions are present.